### PR TITLE
[WNMGDS-658] Move grid styles to utilities

### PR DESCRIPTION
**Pull request by nichia**
Grid classes should work like utilities, where they take higher precedence than components or other classes.

This PR:
- re-group the grid classes in the utilities folder
- Move the grid doc page into utilities as well, similar to USWDS, show at the top of the utilities navigation. 
- Rewrite the layout grid documentation to be more consistent with utility doc page conventions.
- Fixes Github issue: https://github.com/CMSgov/design-system/issues/781

Note: Mgov design system
- To adopt this PR, remove `@import "@cmsgov/design-system/dist/scss/base/grid.scss";` line from file `src/styles/index.scss`

### How to test
- Run the doc site locally with `yarn start`
- Check that `Layout grid` option is grouped at the top of the `Utilities` navigation section, instead of the `Styles` section
- Check the Grid Utility page content is correct
- Check the links/references to Grid utility page is correct
- Test with the "Re-order SCSS imports" PRs on [Hcgov DS](https://github.com/CMSgov/hcgov-design-system/pull/10) and [Mgov DS](https://github.cms.gov/MedicareGov/mgov-design-system/pull/98)
- Check that github issue https://github.com/CMSgov/design-system/issues/781 is fixed

### CODE REVIEW

File: packages/design-system-docs/src/pages/utilities/layout-grid/_grid.docs.scss [hunk 1]
```diff
@@ -12,7 +12,7 @@ The grid consists of three distinct pieces:
 
 Flexbox grid supports modern browsers and Internet Explorer 10+. In older versions of Internet Explorer the grid columns will be stacked.
 
-Style guide: styles.grid
+Style guide: utilities.grid
```
Comment (bernardwang) [inline packages/design-system-docs/src/pages/utilities/layout-grid/_grid.docs.scss:15]: Similar to USWDS, we want to ensure that Layout Grid appears first. I'm not sure if this is currently possible with KSS, so you might want to add that capability by adding this code to `processKssSection`

```
case 'weight':
            section.weight = value;
            break;
```

**Comment by bernardwang** [comment]
`backstop.json` needs to be updated

**Comment by bernardwang** [comment]
There are also quite a few other areas in the documentation to update, just search for `{{root}}/styles/grid/`

File: packages/design-system-docs/src/pages/utilities/layout-grid/_grid.docs.scss (hunk 1 above)
Comment (nichia) [reply packages/design-system-docs/src/pages/utilities/layout-grid/_grid.docs.scss:15]: `processKssSection` doesn't have the `weight` property. I am checking the `section.referenceURI` to update the `section.weight` instead. Think that's workable @bernardwang ?

File: packages/design-system-docs/src/pages/utilities/flexbox/_flexbox.docs.scss [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /*
 Flexbox
 
-Use the [flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) utility classes to control various flexbox properties. These are especially useful when combined with [flexbox grid]({{root}}/styles/grid/#styles.grid.alignment) classes.
+Use the [flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) utility classes to control various flexbox properties. These are especially useful when combined with [flexbox grid]({{root}}//#utilities.grid.alignment) classes.
```
Comment (bernardwang) [inline packages/design-system-docs/src/pages/utilities/flexbox/_flexbox.docs.scss:4]: This doesnt look like the right url

File: packages/design-system-docs/src/pages/utilities/layout-grid/_grid.docs.scss (hunk 1 above)
Comment (bernardwang) [reply packages/design-system-docs/src/pages/utilities/layout-grid/_grid.docs.scss:15]: I was suggesting you add the `weight` property with that code i shared above.

File: packages/design-system-docs/src/pages/utilities/flexbox/_flexbox.docs.scss (hunk 2 above)
Comment (nichia) [reply packages/design-system-docs/src/pages/utilities/flexbox/_flexbox.docs.scss:4]: Missed this, thanks for the pick up.

File: guides/WRITING-DOCUMENTATION.md [hunk 3]
```diff
@@ -76,6 +76,7 @@ To extend the default functionality of KSS, we've implemented support for custom
 | **`@responsive`**           | `@responsive`                                                | Renders breakpoint toggles for the markup example.                                                                                      |
 | **`@status [NAME]`**        | `@status Draft`                                              | Displays a status badge. `[NAME]` must be one of the following values: `Draft`, `Work in progress`, `Ready`, `Deprecated`.              |
 | **`@uswds [URL]`**          | `@uswds https://designsystem.digital.gov/components/button/` | `[URL]` is the link to the corresponding US Web Design System component.                                                                |
+| **`@weight [VALUE]`**       | `@weight -1`                                                 | The weight value controls the order of the page in the left navigation from smallest to largest value.                                  |
```
Comment (bernardwang) [inline guides/WRITING-DOCUMENTATION.md:79]: The weight value controls the order of the page in the navigation from smallest to largest value. When not supplied, the item is sorted alphabetically

File: packages/design-system-docs/src/pages/utilities/layout-grid/_grid.docs.scss [hunk 4]
```diff
@@ -1,30 +1,27 @@
 /*
 Layout grid
 
-The design system provides a 12-column, responsive, flexbox grid to provide structure and align content.
+Use the flexible grid system to structure and align content. The grid is responsive, powered by flexbox, and based on a twelve-column system.
```
Comment (line47) [inline packages/design-system-docs/src/pages/utilities/layout-grid/_grid.docs.scss:4]:
```suggestion
Use the flexible grid system to structure and align content. The grid is responsive, powered by flexbox, and based on a twelve-column grid system.
```

File: packages/design-system-docs/src/pages/utilities/layout-grid/_grid.docs.scss [hunk 5]
```diff
@@ -94,17 +69,39 @@ Align columns horizontally or vertically using [flexbox utility classes]({{root}
...
+
+@responsive
+
+Use a [breakpoint prefix]({{root}}/guidelines/responsive) to change the column widths based on the viewport width.
+
+Responsive format: .ds-l-[breakpoint]-col--*
+
+In the example below, the cells span:
+
+- 1/6 of the row on `xl` viewports, using `.ds-l-xl-col--2`
```
Comment (line47) [inline packages/design-system-docs/src/pages/utilities/layout-grid/_grid.docs.scss:96]: This fraction based terminology is a bit confusing for me. 
What about something like this for these explanations? 

```suggestion
- 6 cells per row on `xl` viewports, using `.ds-l-xl-col--2`
```

**Review by line47 — APPROVED** [review]
A few minor content adjustments, great work Ni!

File: packages/design-system-docs/src/pages/utilities/layout-grid/_grid.docs.scss (hunk 5 above)
Comment (nichia) [reply packages/design-system-docs/src/pages/utilities/layout-grid/_grid.docs.scss:96]: That does read better

**Review by line47 — APPROVED** [review]

**Review by bernardwang — APPROVED** [review]
I tested in both child design systems and in MCT, this looks good to me!

**Pull request merged**